lib/system/gc_common.nim

```diff
@@ -144,15 +144,35 @@ when allowForeignThreadGc:
     ## This function is available only when ``--threads:on`` and ``--tlsEmulation:off``
     ## switches are used
     if not localGcInitialized:
+      if not gch.stackBottom.isNil:
```

Comment:
This is too fragile. Why is it necessary? There is no guarantee that you can call setupForeignThreadGc in a Nim thread.